### PR TITLE
Feature/510 Add F24 to backend

### PR DESCRIPTION
**Pull request by mjtravers**
#510

### CODE REVIEW

**Review by dheitzer — APPROVED** [review]

File: django-backend/fecfiler/reports/form_3x/serializers.py [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+            report_data = get_model_data(validated_data, Report)
+            report_data["form_3x_id"] = form_3x.id
+            report = super().create(report_data)
+            return report
+
+    def update(self, instance, validated_data: dict):
+        with transaction.atomic():
+            for attr, value in validated_data.items():
+                if attr != "id":
+                    setattr(instance.schedule_a, attr, value)
```
Comment (toddlees) [inline django-backend/fecfiler/reports/form_3x/serializers.py:358]: same as above.  should be `setattr(instance.form_3x, attr, value)`

File: django-backend/fecfiler/reports/form_99/serializers.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            report_data = get_model_data(validated_data, Report)
+            report_data["form_99_id"] = form_99.id
+            report = super().create(report_data)
+            return report
+
+    def update(self, instance, validated_data: dict):
+        with transaction.atomic():
+            for attr, value in validated_data.items():
+                if attr != "id":
+                    setattr(instance.schedule_a, attr, value)
```
Comment (toddlees) [inline django-backend/fecfiler/reports/form_99/serializers.py:42]: `setattr(instance.form_99, attr, value)`

File: django-backend/fecfiler/reports/models.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+            # loan guarantors
+            if create_action and self.coverage_through_date:
+                self.pull_forward_loans()
+                self.pull_forward_debts()
+
+    def pull_forward_loans(self):
+        previous_report = self.get_previous_report()
+
+        if previous_report:
+            loans_to_pull_forward = previous_report.transaction_set.filter(
```
Comment (toddlees) [inline django-backend/fecfiler/reports/models.py:91]: we probably only want to pull forward on F3X reports for now and not F24 or F99?

File: django-backend/fecfiler/reports/models.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+                debt.report_id = self.id
+                debt.report = self
+                # The debt_id should point to the original debt transaction
+                # even if the debt is pulled forward multiple times.
+                debt.debt_id = debt.debt_id if debt.debt_id else debt.id
+                self.save_copy(debt)
+
+    def get_previous_report(self):
+        return (
+            Report.objects.get_queryset()
```
Comment (toddlees) [inline django-backend/fecfiler/reports/models.py:144]: probably only F3x for now

File: django-backend/fecfiler/reports/views.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+        return queryset
+
+    @action(
+        detail=True,
+        methods=["post"],
+        url_name="amend",
+    )
+    def amend(self, request, pk):
+        report = self.get_object()
+        report.form_type = "F3XA"
```
Comment (toddlees) [inline django-backend/fecfiler/reports/views.py:118]: not a right now thing, but we'll have to update it for the next type that we can amend

File: django-backend/fecfiler/web_services/dot_fec/dot_fec_composer.py [hunk 6]
```diff
@@ -153,11 +152,11 @@ def compose_dot_fec(report_id, upload_submission_record_id):
         logger.debug(header_row)
         file_content = add_row_to_content(None, header_row)
 
-        f3x_summary = compose_f3x_summary(report_id, upload_submission_record_id)
-        f3x_summary_row = serialize_instance("F3X", f3x_summary)
-        logger.debug("Serialized Report Summary:")
-        logger.debug(f3x_summary_row)
-        file_content = add_row_to_content(file_content, f3x_summary_row)
+        f3x_report = compose_f3x_report(report_id, upload_submission_record_id)
```
Comment (toddlees) [inline django-backend/fecfiler/web_services/dot_fec/dot_fec_composer.py:155]: will deal with in future ticket

**Review by toddlees — CHANGES_REQUESTED** [review]

**Pull request merged**